### ISSUE TITLE
Variables with a mixed primal start values

### ISSUE THREAD

**Issue by rb004f** [JuliaOpt/Gurobi.jl#215]
Problems with variables that have a mix of initial starting points and no initial starting points results in an exception error.  See

```julia
using JuMP
using Gurobi

m = Model()
@variable(m,x, lower_bound=0, upper_bound=2, start=0)
@variable(m,y, lower_bound=0, upper_bound=2)
JuMP.optimize!(m, with_optimizer(Gurobi.Optimizer))
```

This issue is likely related to this issue https://github.com/JuliaOpt/Ipopt.jl/issues/155

**Comment by jd-lara** [comment]
Adding another test to the this thread 

https://gist.github.com/jd-lara/23a6e08a41dc3ad6ee5d239b38c55307

**Comment by odow** [comment]
Solved by the re-write of the MOI wrapper: #216
